File: Programming Basics with JS/T06 - While Loop/exercise/02. Exam Preparation.js
function exam(input) {
    let allowed = Number(input[0]);
    let tasks = 0;
    let badGrades = 0;
    let total = 0;
    let lastTask = '';
    let currentGrade;
    let averageGrade = 0;
    let i = 1;
    let j = 2;

    while (badGrades < allowed) {

        if (input[i] === 'Enough' || input[j] === 'Enough') {
            averageGrade = total / tasks;
            console.log(`Average score: ${averageGrade.toFixed(2)}`);
            console.log(`Number of problems: ${tasks}`);
            console.log(`Last problem: ${lastTask}`);
            return;

        } else {
            currentGrade = Number(input[j]);
            if (currentGrade <= 4.00) {
                badGrades++;
            }
            total += currentGrade;
            tasks++;
            lastTask = input[i];
            i += 2;
            j += 2;
        }
    }
    console.log(`You need a break, ${badGrades} poor grades.`);
}
exam(["3", "Money", "6", "Story", "4", "Spring Time", "5", "Bus", "6", "Enough"]);
exam(["2", "Income", "3", "Game Info", "6", "Best Player", "4"]);

File: Programming Basics with Python/T06 - While Loop/exercise/02. Exam Preparation.py
failed_threshold = int(input())
bad_grades_counter = 0
name_of_last_problem = ""
problems_counter = 0
grades_sum = 0

while True:
    name_of_problem = input()
    if name_of_problem == "Enough":
        break
    grade = int(input())

    if grade <= 4:
        bad_grades_counter += 1

        if bad_grades_counter >= failed_threshold:
            break

    name_of_last_problem = name_of_problem
    problems_counter += 1
    grades_sum += grade

if bad_grades_counter < failed_threshold:
    average_score = grades_sum / problems_counter
    print(f"Average score: {average_score:.2f}")
    print(f"Number of problems: {problems_counter}")
    print(f"Last problem: {name_of_last_problem}")
else:
    print(f"You need a break, {failed_threshold} poor grades.")